### PR TITLE
[FlagGems Operator Development Competition] Add log10, log10_, log10_out operators

### PR DESCRIPTION
**Pull request by iamkarp**
## Summary

This PR implements the `torch.log10` operator for FlagGems as part of the Operator Development Competition (Topic #1).

### Operator Implementation

- **File**: `src/flag_gems/ops/log10.py`
- **Schema**: `log10(Tensor self) -> Tensor`
- **Category**: Pointwise
- **Variants**: `log10`, `log10_` (in-place), `log10_out` (out-of-place)

### Implementation Details

The `log10` operator is implemented using Triton JIT compilation via the `@pointwise_dynamic` decorator. The core computation leverages Triton's native `tl.log2` instruction for optimal GPU performance:

```
log10(x) = log2(x) * log10(2)
```

where `log10(2) = 0.3010299956639812`. This approach uses a single hardware-efficient `log2` instruction rather than a general-purpose `log`, maximizing throughput on GPU hardware.

**Key design choices:**
- Uses `COMPLEX_TO_FLOAT` promotion method consistent with PyTorch's behavior
- Casts input to `tl.float32` for numerical stability before computation
- Follows the same patterns as existing FlagGems operators (`log.py`, `exp2.py`, `abs.py`)

### Registration

- Registered in `src/flag_gems/ops/__init__.py` with all three variants
- Registered in `src/flag_gems/__init__.py` `_FULL_CONFIG` tuple with ATen dispatch names:
  - `"log10"` → `log10`
  - `"log10_"` → `log10_`
  - `"log10.out"` → `log10_out`

### Test Coverage

**Accuracy tests** added to `tests/test_unary_pointwise_ops.py`:

| Test | Shapes | Data Types | Description |
|------|--------|------------|-------------|
| `test_accuracy_log10` | `()`, `(1,)`, `(1024, 1024)`, `(20, 320, 15)`, `(16, 128, 64, 60)`, `(16, 7, 57, 32, 29)` | float16, float32, bfloat16 | Standard forward pass |
| `test_accuracy_log10_` | Same | Same | In-place operation |
| `test_accuracy_log10_out` | Same | Same | Out-of-place with pre-allocated output |
| `test_accuracy_log10_special_values` | Special tensor | Same | Edge cases: 0, 1, 10, 100, 1e-7, inf, nan, negative |

**Accuracy criteria**: Uses `gems_assert_close` with tolerances per the competition specification:
- `float32`: atol = 1.30e-06, rtol = 1e-04
- `float16`: atol = 1.00e-03, rtol = 1e-04
- `bfloat16`: atol = 0.016, rtol = 1e-04

**Performance benchmark** added to `benchmark/test_unary_pointwise_perf.py`:
- Registered as `("log10", torch.log10, FLOAT_DTYPES)` in the forward operations list

### Compatibility

- **Data Types**: Supports float16, float32, bfloat16 (and complex types via promotion)
- **Dimensions**: Tested from 0D scalars through 5D tensors
- **Interface**: Fully aligned with `torch.log10` API (parameter names, input/output formats, default values)

### Files Changed

| File | Change |
|------|--------|
| `src/flag_gems/ops/log10.py` | **New** - Operator implementation |
| `src/flag_gems/ops/__init__.py` | Add log10 imports and exports |
| `src/flag_gems/__init__.py` | Register in `_FULL_CONFIG` for ATen dispatch |
| `tests/test_unary_pointwise_ops.py` | Add accuracy tests (4 test functions) |
| `benchmark/test_unary_pointwise_perf.py` | Add performance benchmark entry |

### CODE REVIEW

**Comment by douxetpur** [comment]
Hi, thank you for your participation in the FlagOS Open Computing Competition 🙏

After review, we have chosen to move forward with a different implementation for this operator. As a result, this PR will be closed.

We appreciate your time and contribution, and hope to see more of your submissions in the future.
